packages/core/web-reporter-ui/src/components/Charts/Chart.tsx

```diff
@@ -74,5 +102,21 @@ export const Chart = ({
 
   const chartOptions = useMemo(() => merge(commonOptions, options), [commonOptions, options]);
 
-  return <ReactApexChart options={chartOptions} series={series} type={type} height={height} />;
+  const ref = useRef<ReactApexChart>(null);
+
+  useEffect(() => {
+    //@ts-expect-error chart is not defined in the typings, but it exists!
+    const chart: ApexCharts | undefined = ref.current?.chart;
+    if (!chart) return;
+
+    toggleSeriesVisibility(
+      chart,
+      series.map((serie) => serie.name).filter((name): name is string => !!name),
+      visibleSeriesNames
+    );
+  }, [series, visibleSeriesNames]);
```

Comment:
the dependency on `series` is likely problematic, when using the live measures, isn't it?

Reply:
It is, but even if I extract the `series.map` part outside of the effect, it will still depend indirectly on `series`.
We could remove the dependency to series altogether but we need to know if it produces unwanted side effects